dspy/teleprompt/signature_opt.py

```diff
@@ -48,6 +48,8 @@ class GenerateInstructionGivenAttempts(dspy.Signature):
 
 class SignatureOptimizer(Teleprompter):
     def __init__(self, prompt_model=None, metric=None, breadth=10, depth=3, init_temperature=1.4, verbose=False, track_stats=False):
+        if breadth <= 1:
```

Comment:
It's fine with one, right? Just has to be positive

Reply:
No, it has to be at least 2. That's what was causing me problems originally, because I set breadth=1 and got weird error messages.